src/merge_sort.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding utf-8 -*-
+
+import timeit
+
+def merge_sort(input_list):
+    if not isinstance(input_list, list):
+        raise TypeError
+
+    if len(input_list) <= 1:
+        return input_list
+
+    list_length = len(input_list)
+    mid = list_length // 2
+    left = merge_sort(input_list[:mid])
+    right = merge_sort(input_list[mid:])
+
+    return merge(left, right)
+
+
+def merge(left, right):
+    ordered_list = []
+    i = 0
```

Comment:
Very good implementation of merge sort! However, it would be better to have more readable and intuitive variable names than `i` and `j`